ADR/0039-spdx-support.md

```diff
@@ -0,0 +1,146 @@
+
+| SPDX Attribute               | CycloneDX Attribute                                           |
+|------------------------------|---------------------------------------------------------------|
+| component.purl = <PURL>      | package.externalRefs = [{referenceCategory:”PACKAGE-MANAGER”, |
+|                              |                          referenceType:purl,                  |
+|                              |                         referenceLocator: <PURL>              |
+|                              |                          }]                                   |
+
+
+#### Component.properties
+CycloneDX components properties describe mapping of string:string properties for given component. SPDX component doesn’t have anything similar to cyclonedx properties. SPDX Package annotations are the only attribute where custom data can be stored and the only “customizable” field where there is comment which is a simple string. Due to that fact, cycloneDX property in format of {“name”: <string>, “value”: <string>} is encoded into json string. There can be also annotations produced by other tools. Therefore to be able to tell annotation comment is json encoded, annotator should ends with string “:jsonencoded”
+
+| SPDX Attribute                  | CycloneDX Attribute                                        |
+|---------------------------------|------------------------------------------------------------|
+| package.properties = [          | component.annotations = [                                  |
+|   {“name”: …, “value”: …}       |   {..., annotator: <tool>”:jsonencoded”                    |
+| ]                               | ]                                                          |
+
+
+#### Formulations
+CycloneDX formulations describe how the container was manufactured. In SPDX, Relationship elements can be used for the same purpose. All elements in SPDX have SPDXID attribute which is an element identifier unique in the whole SBOM document. Relationship element describes relation between two elements using their SPDXID and relationship type. Relationship type BUILD_TOOL_OF can be used to express the relationship of packages which were used to build the container.
+
+| SPDX Attribute                  | CycloneDX Attribute                                        |
+|---------------------------------|------------------------------------------------------------|
+| Formulations.components = [{}]  | Relationships = [{                                         |
+|                                 |  spdxElementId = <A-PACKAGE-ID>                            |
+|                                 |  relationshipType=BUILD_TOOL_OF                            |
+|                                 |  relatedSpdxElement=<ROOT-DOCUMENT-ID>                     |
+|                                 | }]                                                         |
```

Comment:
`<A-PACKAGE-ID> BUILD_TOOL_OF <ROOT-DOCUMENT-ID>` has weird implications. The package is not a build tool for the SPDX document itself (unless the package is cachi2 or syft). It's a build tool for the container we're building.

But I think our SBOMs currently don't include a Component for the container, so this is probably the best we can do for now